clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp

```diff
@@ -10,15 +10,15 @@
 }
 
 #else
-[[cl::reqd_work_group_size(2, 2, 2)]] void not_direct_two() {} // expected-note {{conflicting attribute is here}}
+[[cl::reqd_work_group_size(2, 2, 2)]] void not_direct_two() {} // expected-note 2 {{conflicting attribute is here}}
 
-[[intel::max_work_group_size(1, 1, 1)]] // expected-note {{conflicting attribute is here}}
+[[intel::max_work_group_size(1, 1, 1)]] // expected-note 3 {{conflicting attribute is here}}
```

Comment:
What are the three conflicts?  If I squint, I can see may be two :-)

Reply:
Here is the test output: 
```
/export/iusers/ekeane1/workspaces/sycl/clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp:52:34: error: conflicting attributes applied to a SYCL kernel or SYCL_EXTERNAL function
  parallel_for<class KernelName>([]() {}); // expected-error 3 {{conflicting attributes applied to a SYCL kernel or SYCL_EXTERNAL function}}
                                 ^
/export/iusers/ekeane1/workspaces/sycl/clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp:15:3: note: conflicting attribute is here
[[intel::max_work_group_size(1, 1, 1)]] // expected-note 3 {{conflicting attribute is here}}
  ^
/export/iusers/ekeane1/workspaces/sycl/clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp:13:3: note: conflicting attribute is here
[[cl::reqd_work_group_size(2, 2, 2)]] void not_direct_two() {} // expected-note 2 {{conflicting attribute is here}}
  ^
/export/iusers/ekeane1/workspaces/sycl/clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp:52:34: error: conflicting attributes applied to a SYCL kernel or SYCL_EXTERNAL function
  parallel_for<class KernelName>([]() {}); // expected-error 3 {{conflicting attributes applied to a SYCL kernel or SYCL_EXTERNAL function}}
                                 ^
/export/iusers/ekeane1/workspaces/sycl/clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp:15:3: note: conflicting attribute is here
[[intel::max_work_group_size(1, 1, 1)]] // expected-note 3 {{conflicting attribute is here}}
  ^
/export/iusers/ekeane1/workspaces/sycl/clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp:21:3: note: conflicting attribute is here
[[cl::reqd_work_group_size(4, 4, 4)]] // expected-note 1 {{conflicting attribute is here}}
  ^
/export/iusers/ekeane1/workspaces/sycl/clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp:52:34: error: conflicting attributes applied to a SYCL kernel or SYCL_EXTERNAL function
  parallel_for<class KernelName>([]() {}); // expected-error 3 {{conflicting attributes applied to a SYCL kernel or SYCL_EXTERNAL function}}
                                 ^
/export/iusers/ekeane1/workspaces/sycl/clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp:15:3: note: conflicting attribute is here
[[intel::max_work_group_size(1, 1, 1)]] // expected-note 3 {{conflicting attribute is here}}
  ^
/export/iusers/ekeane1/workspaces/sycl/clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp:13:3: note: conflicting attribute is here
[[cl::reqd_work_group_size(2, 2, 2)]] void not_direct_two() {} // expected-note 2 {{conflicting attribute is here}}
```

It conflicts with 'not_direct_two', func_three, then not-direct-2 does again.  This is a side-effect of us not removing these attributes and leaving them in place.  This is consistent with the existing implementation, except in this case we hit it 2x (instead of the others conflicting with eachother) due to the Depth-first nature.